action.yml

```diff
@@ -80,19 +95,29 @@ runs:
       with:
         node-version: ${{ inputs.node-version }}
 
+    - name: Setup Pages
+      id: pages
+      uses: actions/configure-pages@v3
+
     - name: Install
       shell: "bash"
       run: |
         cd ${{ inputs.path }}
-        $PACKAGE_MANAGER install
+        $PACKAGE_MANAGER $COMMAND
 
     - name: Build
       shell: "bash"
       run: |
         cd ${{ inputs.path }}
-        $PACKAGE_MANAGER run build
+        $RUNNER astro build \
+          --site "${{ steps.pages.outputs.origin }}" \
+          --base "${{ steps.pages.outputs.base_path }}"
```

Comment:
I can understand the hesitation here is it makes customisation of the build step impossible (no way to run pre/post scripts as well for example).

I guess adding an input that tells us the build command risks not working with automatic `--site` and `--base` flags if someone’s package script is not the right form? (e.g. something like `"build": "astro build && node ./post-build"`)

Some options off the top of my head:
- Add `prebuild`/`postbuild` inputs for scripts to run either side of the build
- Add default steps that do this “automatically” (`npm run prebuild --if-present` — maybe no Yarn support?)
- Just add a single build command input that overrides our clever automatic one — up to a user to configure site/base correctly if they decide to override for some reason.

Reply:
Yeah I was checking how Vercel does this, and they phrase it as "overrides," which makes sense. [`actions/configure-pages`](https://github.com/actions/configure-pages) also has an interesting approach where they override the values _in the config file_ after cloning. So that's maybe a more fool-proof method. 

Of course, if Astro accepted `ASTRO_SITE` and `ASTRO_BASE` env variables, that could work too. Or we could just publish a GitHub Pages integration that automatically wires those two values for you?